packages/x-data-grid-pro/src/tests/infiniteLoader.DataGridPro.test.tsx

```diff
@@ -160,4 +160,52 @@ describe('<DataGridPro /> - Infnite loader', () => {
     // should not load more rows because the threshold is not reached
     expect(getRow.callCount).to.equal(5);
   });
+
+  it('should not call `onRowsScrollEnd` if there are rows pinned to the bottom and the viewport scroll is at the top', async function test() {
```

Comment:
This test seems to be flaky – it failed in https://github.com/mui/mui-x/pull/14613
https://app.circleci.com/pipelines/github/mui/mui-x/66642/workflows/92f1e33d-ab21-44e8-b769-1b0dc14cade0/jobs/380114

Reply:
I assume it is because of `sleep`. Is it fine if I add the fix to your PR?

Reply:
Sure 👍🏻

Reply:
Failed on this [one](https://github.com/mui/mui-x/pull/14124) too

Reply:
I dropped the `sleep` because if it's called twice the test will fail anyway.